assignments/02-collections-lifetimes/src/life.rs

```diff
@@ -1,19 +1,106 @@
-fn split_string(string: &str, delimeter: &str) -> Vec<&str> {
-    todo!()
+#[allow(dead_code)]
+pub fn split_string<'a>(string: &'a str, delimeter: &str) -> Vec<&'a str> {
+    if string.is_empty() {
+        vec![]
+    } else {
+        let output: Vec<&str> = string.split_terminator(delimeter).collect();
+        output
+    }
 }
 
 #[derive(PartialEq, Debug)]
-struct Differences {
-    only_in_first: Vec<&str>,
-    only_in_second: Vec<&str>,
+#[allow(dead_code)]
+struct Differences<'a> {
+    only_in_first: Vec<&'a str>,
+    only_in_second: Vec<&'a str>,
```

Comment:
This works, but they don't necessarily have to be in the same lifetime (one can be 'a & another 'b)